extension/src/popup/components/manageAssets/ManageAssetRows/index.tsx

```diff
@@ -279,78 +314,122 @@ export const ManageAssetRows = ({
-                    isLoading={
-                      isActionPending && assetSubmitting === canonicalAsset
-                    }
-                    onClick={() => {
-                      if (isContract) {
-                        handleTokenRowClick(issuer, canonicalAsset);
-                      } else {
-                        handleRowClick(
-                          { code, issuer, image, domain },
-                          isTrustlineActive,
-                        );
+          {assetRows.map(
+            ({ code = "", domain, image = "", issuer = "", name = "" }) => {
+              if (!accountBalances.balances) return null;
+              const isContract = isContractId(issuer);
+              const canonicalAsset = getCanonicalFromAsset(code, issuer);
+              const isTrustlineActive =
+                Object.keys(accountBalances.balances).some(
+                  (balance) => balance === canonicalAsset,
+                ) || accountBalances.tokensWithNoBalance.includes(issuer);
+              const isActionPending =
+                submitStatus === ActionStatus.PENDING ||
+                accountBalanceStatus === ActionStatus.PENDING;
+              return (
+                <div
+                  className="ManageAssetRows__row"
+                  key={canonicalAsset}
+                  data-testid="ManageAssetRow"
+                >
+                  <ManageAssetRow
+                    code={code}
+                    issuer={issuer}
+                    image={image}
+                    domain={domain}
+                    name={name}
+                  />
+                  <div className="ManageAssetRows__button">
+                    <Button
+                      size="md"
+                      variant="secondary"
+                      disabled={isActionPending}
+                      isLoading={
+                        isActionPending && assetSubmitting === canonicalAsset
                       }
-                    }}
-                    type="button"
-                    data-testid="ManageAssetRowButton"
-                  >
-                    {isTrustlineActive || isContract ? t("Remove") : t("Add")}
-                  </PillButton>
+                      onClick={() => {
+                        if (isContract) {
+                          handleTokenRowClick(
+                            { code, issuer, image, domain },
+                            isTrustlineActive,
+                            canonicalAsset,
+                          );
+                        } else {
+                          handleRowClick(
+                            { code, issuer, image, domain },
+                            isTrustlineActive,
+                          );
+                        }
+                      }}
+                      type="button"
+                      data-testid="ManageAssetRowButton"
+                    >
+                      {isTrustlineActive ? (
```

Comment:
Should this still be `isTrustlineActive || isContract`?

Reply:
Ah, good callout. I actually refactored `isTrustlineActive` up above to include all assets in `balances` (classic or token) and `tokensWithNoBalance`, so this should cover all asset/tokens a user has associated with their account. (at least in my testing it does 😬 )